### DISCUSSION TITLE
Remix

### DISCUSSION THREAD

**Discussion by jdmann** [TanStack/query#3211] [Q&A]
For apps already using React Query but also want to implement Remix, what is the recommended approach for handling queries? Does anything need to be configured differently in order to make optimal use of the Remix framework?

**Comment by cliffordfajardo** [comment]
I'm also fairly new to the Remix world 💿  and have been curious about this topic area too

EDIT: 
Examples of React-Router v6.5 + react-query: https://tkdodo.eu/blog/react-query-meets-react-router


## Community Chatter on Remix with `react-query`
_**Note: I would dig deeper into these assumptions, but I haven't had  the chance to form strong opinions on the thoughts below:**_

Kent C Dodds
> you don't even need react-query/Apollo/redux or even axios for making network requests. It's all handled for you + much more by the framework. **To be clear, you can totally use these tools with Remix and the fact that you use and love them is reason enough to bring them along for the party**_ - [twitter thread discussion](https://mobile.twitter.com/kentcdodds/status/1463994662281048074)


## Other resources
- [`remix-query` a small library](https://github.com/GuptaSiddhant/remix-query), about [180 lines of code](https://github.com/GuptaSiddhant/remix-query/blob/main/src/useQuery.ts#L10) &  API is inspired by react-query

Thanks for starting this discussion 🙏 
CC @TkDodo - i think you might find it interesting; have you formed any thoughts on Remix 📀 ?

**Comment by WesleyYue** [comment]
There's also a bunch of good information here: https://tanstack.com/query/v4/docs/react/guides/ssr#using-remix

**Reply by cliffordfajardo** [reply]
Also Remix is currently working on `clientLoader` and `clientAction` API that should make it easier to integrate with react-query easier 🤞 

This API was initially sparked from this discussion in the Remix repo:
[Optional Client Side Memory Caching Support #4950](https://github.com/remix-run/remix/discussions/4950), which brough to life this work being done by the team here "[initial work on clientLoader & clientAction #5543](https://github.com/remix-run/remix/pull/5543)"

**Reply by thephfernandes** [reply]
Looking forward to this!

**Comment by hanihusam** [comment]
Any update on how Remix SPA mode meets React Query?

**Reply by joernroeder** [reply]
this

**Reply by anth0ni3** [reply]
Any news??

**Reply by TkDodo** [reply]
how is remix spa mode different from react-router ?

**Reply by Gallevy** [reply]
@TkDodo because of the clientLoader pattern, but then it resemble more the SSR remix way with loader but with unneccessary overhead of having to need to link useLoaderData with initialData of the query

**Reply by devo-devo** [reply]
Would love to hear if any one has had any luck/experience integrating SPA mode with react-query. I'll try follow the previous react-router + query examples [here](https://codesandbox.io/s/github/tanstack/query/tree/main/examples/react/react/react-router?embed=1&theme=light&file=src/index.tsx) and [here](https://tkdodo.eu/blog/react-query-meets-react-router) and report back....

**Reply by stevan-borus** [reply]
@devo-devo https://x.com/stevanborus/status/1802761540220645508

**Reply by devo-devo** [reply]
Great work @stevan-borus! Beautifully presented. Thank you.

**Reply by kasvith** [reply]
would be good to have this as an example
